server/src/omtDocumentInformationProvider.ts

```diff
@@ -0,0 +1,359 @@
+    return findUsagesInLine(fileImports.map(x => x.name), l, line);
+}
+
+function getLocalLocationsForCode(declaredObjects: OmtLocalObject[], l: number, line: string): OmtLocalObject[] {
+    return findUsagesInLine(declaredObjects.map(x => x.name), l, line);
+}
+
+function findUsagesInLine(declaredObjects: string[], l: number, line: string): OmtLocalObject[] {
+    const documentLinks: OmtLocalObject[] = [];
+
+    declaredObjects.filter(declaredObject => line.includes(declaredObject)).forEach(declaredObject => {
+        const characterIndex = line.search(new RegExp(`${declaredObject}(?=[^a-zA-Z0-9]|$)`)) ?? line.indexOf(declaredObject);
+
+        if (characterIndex >= 0) {
+            documentLinks.push({
+                name: declaredObject,
+                range: Range.create({ line: l, character: characterIndex }, { line: l, character: characterIndex + declaredObject.length })
+            });
+        }
+    });
+    return documentLinks;
+}
+
+export function getAvailableObjectsFromDocument(document: TextDocument, shorthands?: Map<string, string>): OmtAvailableObjects {
+    const documentText = document.getText();
+    const yamlDocument = parse(document.getText());
+    const availableImports: OmtImport[] = [];
+    const definedObjects: OmtLocalObject[] = [];
+    if ("import" in yamlDocument && shorthands) {
+        const documentImports = yamlDocument["import"];
+        const importUrls = Object.keys(documentImports);
+        importUrls.forEach(importUrl => {
+            const imports: string[] = documentImports[importUrl];
+            const uriMatchResult = getUriMatch(' ' + importUrl, document, shorthands);
+            if (uriMatchResult) {
+                imports.forEach(importName => {
+                    availableImports.push({ name: importName, url: importUrl, fullUrl: uriMatchResult.url });
+                });
+            }
+        });
+    }
+    if ("queries" in yamlDocument) {
+        definedObjects.push(...findDefinedObjects(yamlDocument["queries"], documentText, "QUERY"));
+    }
+    if ("commands" in yamlDocument) {
+        definedObjects.push(...findDefinedObjects(yamlDocument["commands"], documentText, "COMMAND"));
+    }
+    if ("model" in yamlDocument) {
+        const modelEntries = yamlDocument["model"];
+        definedObjects.push(...findModelEntries(modelEntries, documentText));
+        const keys = Object.keys(modelEntries);
+        keys.forEach(key => {
+            const entry = modelEntries[key];
+
+            if ("commands" in entry) {
+                definedObjects.push(...findDefinedObjects(entry["commands"], documentText, "COMMAND"));
+            }
+
+            if ("queries" in entry) {
+                definedObjects.push(...findDefinedObjects(entry["queries"], documentText, "QUERY"));
+            }
+        });
+    }
+    return { availableImports, definedObjects };
+}
+
+function findModelEntries(modelEntries: any, documentText: string): OmtLocalObject[] {
```

Comment:
This any gives rise to compilation warnings, do you thing we can make this into an `object` maybe?

Reply:
The result of `parse(document.getText())` is `any` as you can see here: https://eemeli.org/yaml/v1/#parse-amp-stringify. I will check if I can use `Object` instead.